### PR TITLE
HPCC-29301 Guard against potential (theoretical) use of null pointer

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: system/jhtree/jhinplace.cpp [hunk 1]
```diff
@@ -2507,21 +2507,25 @@ void CInplaceLeafWriteNode::write(IFileIOStream *out, CRC32 *crc)
...
+    if (helper)
     {
-        //Create a representation of the null row - potentially used for the new compression algorithm
-        byte * nullRow = new byte[keyedSize];
-        RtlStaticRowBuilder rowBuilder(nullRow, keyedSize);
-
-        auto & meta = format->queryRecordAccessor(true);
-        size32_t offset = 0;
-        for (unsigned idx = 0; idx < meta.getNumFields() && offset < keyedSize; idx++)
+        //Nothing currently relies on the null row being created....
```
Comment (AttilaVamos) [inline system/jhtree/jhinplace.cpp:2512]: Are this and the comment in line 2516 controversial?

File: system/jhtree/jhinplace.cpp [hunk 2]
```diff
@@ -2507,21 +2507,25 @@ void CInplaceLeafWriteNode::write(IFileIOStream *out, CRC32 *crc)
...
         {
-            const RtlFieldInfo *field = meta.queryField(idx);
-            offset = field->type->buildNull(rowBuilder, offset, field);
+            //Create a representation of the null row - potentially used for the new compression algorithm
+            byte * nullRow = new byte[keyedSize];
+            RtlStaticRowBuilder rowBuilder(nullRow, keyedSize);
+
+            auto & meta = format->queryRecordAccessor(true);
+            size32_t offset = 0;
+            for (unsigned idx = 0; idx < meta.getNumFields() && offset < keyedSize; idx++)
```
Comment (AttilaVamos) [inline system/jhtree/jhinplace.cpp:2522]: Is this loop a kind of (missing) initialisation?

**Review by AttilaVamos — COMMENTED** [review]
Seems OK, but there are two questions.

File: system/jhtree/jhinplace.cpp (hunk 1 above)
Comment (ghalliday) [reply system/jhtree/jhinplace.cpp:2512]: no, was used in the original code, and could theoretically be used in the future.

File: system/jhtree/jhinplace.cpp (hunk 2 above)
Comment (ghalliday) [reply system/jhtree/jhinplace.cpp:2522]: I don't understand the question, what do you mean?

Comment (AttilaVamos) [reply system/jhtree/jhinplace.cpp:2522]: Sorry, I was lost in the changes and how GitHub is presenting them.

**Review by AttilaVamos — COMMENTED** [review]
Seems good to merge.

**Review by AttilaVamos — APPROVED** [review]

**Pull request merged**